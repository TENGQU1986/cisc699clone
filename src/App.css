* {
    box-sizing: border-box;
}

.App {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    margin-bottom: 50px;
}

.Child {
    border: 1px solid;
    margin: 20px;
    height: 100px;
    width: 100vw;
    max-width: 800px;
}

.parent {
    border: 1px solid;
    margin: 20px;
    width: 100vw;
    max-width: 800px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-start;
    padding-left: 50px;
    padding-bottom: 20px;
}
.button {
    background-color: #1c88bf;
    padding: 0.5rem;
    color: #fff;
    font-size: 1.2rem;
    border: none;
    font-weight: 300;
    margin-top: 1.2rem;
}
.container {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-start;
    width: 100vw;
    max-width: 800px;
    margin-left: 200px;
}
.cache-header { 
    padding-left: 260px;
    display: flex;
}
/* .lazy {
    width: 100vw;
    max-width: 800px;
} */
.lazy_container {
    max-width: 800px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}
.lazyLoading {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-start;
    margin-bottom: 50px;
    margin-left: 200px;
}
.lazyLoading-title {
    padding-left: 260px;
    display: flex;
}


