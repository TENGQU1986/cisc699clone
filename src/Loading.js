import React from "react";
import loading from "./images/loader.gif";

const Loading = () => {
  return (
    <div>
      <img src={loading} alt="" height={200} width={200} />
    </div>
  );
};

export default Loading;
