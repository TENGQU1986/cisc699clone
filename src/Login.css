.box-layout {
    display: flex;
    align-items: center;
    background: url('/src/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    /* display: flex;
    flex-direction: column;
    justify-content: space-around; */
    background: white;
    border-radius: 3px;
    padding: 3.2rem 1.6rem;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 1.6rem 0;
    line-height: 1;
    
}

.login-button {
    background-color: #1c88bf;
    padding: 1.2rem;
    color: #fff;
    font-size: 1.8rem;
    border: none;
    font-weight: 300;
    margin-top: 1.2rem;
}