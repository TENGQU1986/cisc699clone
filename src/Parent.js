import React, { useState, useEffect } from "react";
import Child from "./Child";
import ChildWithoutMemo from "./ChildWithoutMemo";
import "./App.css";
import Comments from "./Comments";
import LazyLoading from "./LazyLoading";

const Parent = ({ user }) => {
  const [counter, setCounter] = useState(0);
  const [input, setInput] = useState("");
  const [displayMemo, setDisplayMemo] = useState(false);
  const list = [1, 2, 3];

  useEffect(() => {
  }, []);

  const handleCounterClick = () => {
    setCounter(counter + 1);
  };

  const onInputChange = (e) => {
    const value = e.target.value;
    setInput(value);
  };

  const handleSwitch = () => {
    setDisplayMemo(!displayMemo);
  };
  return (
    <div className="App">
      <h1>Rendering Times Demo</h1>
      <span>hint: you can open developer tools to see the differentces</span>
      <div className="parent">
        <button onClick={handleSwitch} title="parent_button" className="button">
          {displayMemo ? "switch to unmemo" : "switch to memo"}
        </button>
        <h1>Hello from Parent: {counter}</h1>
        <div>
          <button onClick={handleCounterClick}>+1</button>
          <input
            type="text"
            onChange={onInputChange}
            title="parent_input"
            className="input"
          />
        </div>
      </div>
      {displayMemo ? (
        <Child counter={counter} list={list} />
      ) : (
        <ChildWithoutMemo counter={counter} list={list} />
      )}

    </div>
  );
};

export default Parent;
