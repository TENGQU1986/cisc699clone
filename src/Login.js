import React from "react";
import { auth, provider } from "./firebase/firebase";
import './Login.css';

const Login = () => {
  const signIn = () => {
    auth.signInWithPopup(provider).catch(alert);
  };
  return (
    <div>
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">React Performance</h1>
          <span>Please login to share your thoughts.</span>
          <button onClick={signIn} className="login-button">
            Login with Google
          </button>
        </div>
      </div>
    </div>
  );
};

export default Login;
