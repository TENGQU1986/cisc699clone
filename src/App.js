import React, { useState } from "react";
import { useAuthState } from "react-firebase-hooks/auth";
import { auth, db, provider } from "./firebase/firebase";
import "./App.css";
import Login from "./Login";
import Parent from "./Parent";
import Loading from "./Loading";
import LazyLoading from "./LazyLoading";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import Header from "./Header";

import CssBaseline from "@material-ui/core/CssBaseline";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import Comments from "./Comments";
import ServerSide from "./ServerSide";
import Demo from "./Demo";
import ClientSide from "./ClientSide";

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
});

const App = () => {
  const [user, loading] = useAuthState(auth);
  const classes = useStyles();
  const [value, setValue] = React.useState(0);
  const [step, setStep] = useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };
  if (!user) return <Login />;
  if (loading) return <Loading />;

  const handleStepChange = (e) => {
    if (e.target.innerHTML === 'Server-side') {
      setStep(0)
    } else if (e.target.innerHTML === 'Client-side') {
      setStep(1)
    } else if (e.target.innerHTML === 'Demo') {
      setStep(2)
    }
  }

  return (
    <div>
      <Header user={user} />
      <CssBaseline />
      <Container maxWidth="lg">
        {/* <Typography component="div" style={{ backgroundColor: '#cfe8fc', height: '100vh' }} /> */}

        <Paper className={classes.root}>
          <Tabs
            value={value}
            onChange={handleChange}
            indicatorColor="primary"
            textColor="primary"
            centered
          >
            <Tab label="Server-side" onClick={handleStepChange}/>
            <Tab label="Client-side" onClick={handleStepChange}/>
            <Tab label="Demo" onClick={handleStepChange}/>
          </Tabs>
        </Paper>
        {/* <Parent user={user} /> */}
        {
          step === 0 ? (<ServerSide />) : (
            step === 1 ? (<ClientSide />) : (<Demo user={user} />)
          ) 
        }
        <Comments />
      </Container>
    </div>
  );
};

export default App;
