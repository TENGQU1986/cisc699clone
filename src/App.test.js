import { render, fireEvent, act } from "@testing-library/react";

import App from "./App";
import Loading from "./Loading";
import Login from "./Login";
import Parent from "./Parent";

describe("should render App", () => {
  let user;
  let loading;
  it("should render login page when no user", () => {
    user = null;

    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
    
  });

  it("should render Loading page when no user and loading is true", () => {
    user = 'some user';
    loading = true;

    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot()
  });

  it('should render Parent component when there is user', () => {
      user = 'some user';
      loading = false;
      const { container } = render(<Parent />);
      expect(container).toMatchSnapshot()
  })
});
