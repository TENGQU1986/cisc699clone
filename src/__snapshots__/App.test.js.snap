// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App should render Loading page when no user and loading is true 1`] = `
<div>
  <div>
    <img
      alt=""
      height="200"
      src="loader.gif"
      width="200"
    />
  </div>
</div>
`;

exports[`should render App should render Parent component when there is user 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Rendering Times Demo
    </h1>
    <span>
      hint: you can open developer tools to see the differentces
    </span>
    <div
      class="parent"
    >
      <button
        class="button"
        title="parent_button"
      >
        switch to memo
      </button>
      <h1>
        Hello from Parent: 
        0
      </h1>
      <div>
        <button>
          +1
        </button>
        <input
          class="input"
          title="parent_input"
          type="text"
        />
      </div>
    </div>
    <div
      class="Child"
    >
      <h1>
        Child without Memoization - 
        0
      </h1>
    </div>
  </div>
</div>
`;

exports[`should render App should render login page when no user 1`] = `
<div>
  <div>
    <div
      class="box-layout"
    >
      <div
        class="box-layout__box"
      >
        <h1
          class="box-layout__title"
        >
          React Performance
        </h1>
        <span>
          Please login to share your thoughts.
        </span>
        <button
          class="login-button"
        >
          Login with Google
        </button>
      </div>
    </div>
  </div>
</div>
`;
